src/shared/mount-util.c

```diff
@@ -1894,7 +1894,7 @@ static int path_get_mount_info_at(
         if (r < 0)
                 return log_debug_errno(r, "Failed to get mount ID: %m");
 
-        r = libmount_parse("/proc/self/mountinfo", NULL, &table, &iter);
+        r = libmount_parse(NULL, NULL, &table, &iter);
```

Comment:
oh, ah.

maybe prefix the two NULL args with the usual `/* path= */` and  `/* source= */`

Reply:
might be a good idea to add a comment here, that by passing NULL/NULL we make sure to get utab loaded, and thus get userspace options too

Reply:
you could even be even smarter and set the first arg as before as long as ret_options is NULL, after all if we don't need the options we don't need utab.

Reply:
I think we should systematically take utab into account, to avoid pitfalls like this and to avoid hardcoding /p/s/mountinfo unnecessarily.

Reply:
Probably a wrapper around libmount_parse then?

`int libmount_parse_mountinfo(bool with_utab, ...)`

Reply:
Could you elaborate more? I cannot follow, sorry.

Reply:
Instead of implying this bit of info in whether /p/s/mountinfo is specified or not let's instead turn it into a proper bool?

Reply:
OK. libmount_parse_mountinfo() and libmount_parse_with_utab() are introduced. PTAL.